eng/pipelines/templates/stages/platform-matrix.json

```diff
@@ -1,6 +1,7 @@
 {
   "displayNames": {
-    "true": "TestFromSource"
+      "false": "",
```

Comment:
What cases would we need to nullify false in the display name?

Reply:
I think I understand that you are doing here now. Why not use a similar pattern to TestFromSourceConfig and next the coverage option? While this mapping works right now if we ever have any other true/false values this might map them as well.

Reply:
The problem is this value is made to be overridden/updated by different test scenarios. And due to the "grouping name" problem, it will list the job display name as "Coverage" regardless of whether the value is false or true.

Looks like this PR may not be the best attempt regardless, so a moot point here, but this is an annoying side effect of auto-name generation+using a simple config with less metadata support per field. I still don't have a great solution.

Reply:
I wonder if we can come up with some sort of display name format string. Not sure if it will work or handle all the cases, or really deal with conflicts but it is just me brainstorming.

Reply:
Hmm, riffing on that idea, I wonder if adding support for JSON comments would give us a a convenient place to do code-gen hinting and formatting without over-complicating the config schema, e.g.

```
// use key
{
  "matrix": {
    "Coverage": "true"  //+matrix:displayName:$key
  }
}

// use literal
{
  "matrix": {
    "Coverage": "true"  //+matrix:displayName:EnableCoverage
  }
}

// use formatted string with key and value
{
  "matrix": {
    "Coverage": "true"  //+matrix:displayName:$key_$value
  }
}
```

Reply:
An alternative more along what you suggested:

```
// Replace display name for Coverage variable with dynamic format string
{
  "displayNames": {
    "Coverage": "$key_$value"
  },
  "matrix": {
    "Coverage": "true"
  }
}

```

Though I've always felt like `displayNames` as a top-level key was somewhat out of place and ruined the aspect of the config where every key had a direct relationship to the actual matrix, as opposed to an auxiliary purpose.

Reply:
We can keep thinking about the display name outside the scope of this PR. I can live with what you have here for this case.